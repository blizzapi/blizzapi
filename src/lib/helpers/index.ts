export default {
  query: (endpoint: string) => {
    console.log(`query: ${endpoint}`);
  }
}