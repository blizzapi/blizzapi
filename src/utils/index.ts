export * from './localization';
export *  from './api';
export *  from './oauth';
