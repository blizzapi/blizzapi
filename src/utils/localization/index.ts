export * from "./regions";
export * from "./locales";
export * from "./defaultLocales";
export * from "./sc2Realms";
