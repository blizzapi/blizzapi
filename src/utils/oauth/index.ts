export * from "./authorizeUris";
export * from "./checkTokenUris";
export * from "./tokenUris";
