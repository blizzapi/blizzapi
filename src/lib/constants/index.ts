/**
 * Class exposing configuration constants
 *
 * Via:
 *
 * https://develop.battle.net/documentation/guides/using-oauth
 * https://develop.battle.net/documentation/guides/regionality-partitions-and-localization
 */

import * as localization from './localization';
import * as api from './api';
import * as oauth from './oauth';

const CONSTANTS = {
  REGIONS: localization.regions,
  LOCALES: localization.locales,
  DEFAULT_LOCALES: localization.defaultLocales,
  SC2_REALMS: localization.sc2Realms,
  REGION_API_HOSTS: api.regionHosts,
  OAUTH_AUTHORIZE_URIS: oauth.authorizeUris,
  OAUTH_TOKEN_URIS: oauth.tokenUris,
  OAUTH_CHECK_TOKEN_URIS: oauth.checkTokenUris,
};

export default CONSTANTS;
