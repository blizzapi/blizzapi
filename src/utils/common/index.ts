import constants from '../../constants';
import {
  validateRegionId,
  validateRegionName,
  getRegionIdByName,
} from '../localization/regions';
import {
  ConstantKey,
  RegionIdOrName,
  RegionIdAsNumberOrString,
  RegionName,
} from '../../types';

export const getConstantByRegionId = (
  regionId: RegionIdAsNumberOrString,
  constantKey: ConstantKey,
) => {
  const regionIdAsString = typeof regionId !== 'string'
    ? regionId
    : regionId.toString();
  const isRegionIdValid = validateRegionId(regionIdAsString);

  if (!isRegionIdValid) {
    throw new RangeError(
      `${regionIdAsString} is not a valid parameter for getConstantByRegionId(${regionId}, '${constantKey}')`,
    );
  }
  return constants[constantKey][regionIdAsString];
};

/**
 * Returns constant value for given region name from specified constant object
 *
 * @param regionName Region name as two-letter string (e.g. 'us')
 * @param constantKey Constant key
 * @return Uri as a string
 */
export const getConstantByRegionName = (regionName: RegionName, constantKey: ConstantKey) => {
  const isRegionNameValid = validateRegionName(regionName);

  if (!isRegionNameValid) {
    throw new RangeError(
      `${regionName} is not a valid parameter for getConstantByRegionName(${regionName}, '${constantKey}')`,
    );
  }

  const regionId = getRegionIdByName(regionName);

  return constants[constantKey][regionId];
};

export const getConstantByRegion = (regionIdOrName: RegionIdOrName, constantKey: ConstantKey) =>
  (validateRegionId(regionIdOrName)
    ? getConstantByRegionId(regionIdOrName, constantKey)
    : getConstantByRegionName(regionIdOrName.toString(), constantKey));
