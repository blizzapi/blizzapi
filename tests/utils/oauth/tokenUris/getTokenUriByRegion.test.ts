import * as utils from '../../../../src/lib/utils';

const { getTokenUriByRegion } = utils;

import regionNamesJson from '../../../__testData__/regionNames.json';
import wrongRegionNamesJson from '../../../__testData__/wrongRegionNames.json';
import regionIdsJson from '../../../__testData__/regionIds.json';
import wrongRegionIdsJson from '../../../__testData__/wrongRegionIds.json';

/* tslint:disable no-expression-statement */
describe('getTokenUriByRegion()', () => {
  test('should be defined', () => {
    expect(getTokenUriByRegion).toBeDefined();
  });

  test('should be of type "function"', () => {
    expect(typeof getTokenUriByRegion).toBe('function');
  });

  (regionIdsJson as ReadonlyArray<string | number>).forEach(regionId => {
    test(`should return correct value for ${regionId} as valid region id`, () => {
      expect(getTokenUriByRegion(regionId)).toMatchSnapshot();
    });
  });

  (wrongRegionIdsJson as ReadonlyArray<string | number>).forEach(wrongRegionId => {
    test(`should throw RangeError for ${wrongRegionId} as invalid parameter`, () => {
      expect(() => getTokenUriByRegion(wrongRegionId)).toThrow(RangeError);
    });
  });

  (regionNamesJson as ReadonlyArray<string>).forEach(regionName => {
    test(`should return correct value for ${regionName} as valid region name`, () => {
      expect(getTokenUriByRegion(regionName)).toMatchSnapshot();
    });
  });

  (wrongRegionNamesJson as ReadonlyArray<string>).forEach(wrongRegionName => {
    test(`should throw RangeError for ${wrongRegionName} as invalid region name`, () => {
      expect(() => getTokenUriByRegion(wrongRegionName)).toThrow(RangeError);
    });
  });
});
/* tslint:enable no-expression-statement */
