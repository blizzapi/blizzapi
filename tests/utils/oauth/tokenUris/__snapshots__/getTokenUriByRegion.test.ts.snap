// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTokenUriByRegion() should return correct value for 1 as valid region id 1`] = `"https://us.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for 1 as valid region id 2`] = `"https://us.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for 2 as valid region id 1`] = `"https://eu.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for 2 as valid region id 2`] = `"https://eu.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for 3 as valid region id 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for 3 as valid region id 2`] = `"https://apac.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for 5 as valid region id 1`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for 5 as valid region id 2`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for CN as valid region name 1`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for Cn as valid region name 1`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for EU as valid region name 1`] = `"https://eu.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for Eu as valid region name 1`] = `"https://eu.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for KR as valid region name 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for Kr as valid region name 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for TW as valid region name 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for Tw as valid region name 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for Us as valid region name 1`] = `"https://us.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for cN as valid region name 1`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for cn as valid region name 1`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for eU as valid region name 1`] = `"https://eu.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for eu as valid region name 1`] = `"https://eu.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for kR as valid region name 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for kr as valid region name 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for tW as valid region name 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for tw as valid region name 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for uS as valid region name 1`] = `"https://us.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for us as valid region name 1`] = `"https://us.battle.net/oauth/token"`;

exports[`getTokenUriByRegion() should return correct value for us as valid region name 2`] = `"https://us.battle.net/oauth/token"`;
